File: heart/src/main/java/com/vantagecircle/heartrate/activity/ActivityComponent.java
package com.vantagecircle.heartrate.activity;

import com.vantagecircle.heartrate.AppComponent;
import com.vantagecircle.heartrate.activity.ui.HeartActivity;
import com.vantagecircle.heartrate.activity.ui.WelcomeActivity;
import com.vantagecircle.heartrate.scope.PerActivity;

import dagger.Component;

/**
 * Created by bapidas on 14/11/17.
 */
@PerActivity
@Component(dependencies = AppComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {
    void inject(WelcomeActivity welcomeActivity);
    void inject(HeartActivity heartActivity);
}

File: app/src/main/java/com/bapidas/heartrate/di/component/ActivityComponent.kt
package com.bapidas.heartrate.di.component

import com.bapidas.heartrate.di.annotation.PerActivity
import com.bapidas.heartrate.di.module.ActivityModule
import com.bapidas.heartrate.ui.activity.ui.HeartActivity
import dagger.Component

/**
 * Created by bapidas on 14/11/17.
 */
@PerActivity
@Component(dependencies = [AppComponent::class], modules = [ActivityModule::class])
interface ActivityComponent {
    fun inject(heartActivity: HeartActivity?)
}